File: src/main/java/com/mini/credit/dto/credit/PaiementInitialDemandeCreditRequest.java
package com.mini.credit.dto.credit;

import com.mini.credit.enums.ModePaiement;
import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Getter
@Setter
public class PaiementInitialDemandeCreditRequest {

    @NotNull
    private Long sessionCaisseId;

    @NotNull
    private Long caisseId;

    private Long agentId;

    private Long createdById;

    @NotNull(message = "La date du paiement est obligatoire")
    private LocalDateTime datePaiement;

    @NotNull
    private ModePaiement modePaiement;

    @DecimalMin(value = "0.00")
    private BigDecimal fraisPayes = BigDecimal.ZERO;

    @DecimalMin(value = "0.00")
    private BigDecimal depotGarantiePaye = BigDecimal.ZERO;

    private String observation;
}

File: frontend/src/app/features/credit/models/paiement-initial-demande-credit-request.ts
import { ModePaiement } from "../../../shared/enums/mode-paiement.enum";

export interface PaiementInitialDemandeCreditRequest {
  sessionCaisseId?: number;
  caisseId?: number;

  agentId?: number;
  createdById?: number;

  datePaiement?: string;

  modePaiement: ModePaiement;

  fraisPayes: number;
  depotGarantiePaye?: number;

  observation?: string;
}